### DISCUSSION TITLE
help a novice. connect a mouse the pico to a computer

### DISCUSSION THREAD

**Discussion by AbhishekAKA47** [sekigon-gonnoc/Pico-PIO-USB#73] [Q&A]
![WhatsApp Image 2023-03-28 at 12 16 59 AM](https://user-images.githubusercontent.com/127853293/228037716-6e745dda-7381-433c-9646-8cab3ef51996.jpeg)I want to connect the mouse to the pico and the pico to the computer at the same time but don't know what hardware I will need and which steps I will have to take. i just don't know the hardware side of this I think I have uploaded the correct code using the Arduino ide, just missing how to connect the mouse to the pico and the pico to the computer. please help.

**Comment by ccrome** [comment]
By default, D+ is on GP0 and D- is GP1. 

so, the connection should look like this:
![image](https://user-images.githubusercontent.com/286973/228044538-13d7d806-08d3-40cd-9f67-dd129ae32707.png)

You likely have a difference USB A socket.  Just make sure you connect VBUS to VBUS, GND to GND, GP0 to D+ and GP1 to D-.

**Reply by AbhishekAKA47** [reply]
in this do I just upload the https://github.com/sekigon-gonnoc/Pico-PIO-USB/tree/main/examples/capture_hid_report code using arduino ide and then connect my mouse to the usbafemale and micro usb to the computer? should that turn the pico into a man in the middle? 
in resource usage it says 'Two GPIO for D+/D- (Series 22ohm resistors are better)' do I have to connect these in between the usbafemale and gpio pins? 
also on this other persons pico https://github.com/No0ne/ps2x2pico they use bi-directional level shifter

**Reply by ccrome** [reply]
> in this do I just upload the https://github.com/sekigon-gonnoc/Pico-PIO-USB/tree/main/examples/capture_hid_report code using arduino ide and then connect my mouse to the usbafemale and micro usb to the computer? 

That goes from USB mouse -> serial port.  So, open the serial port that's created when you plug it in.  It's man in the middle, but not mouse->mouse.


> should that turn the pico into a man in the middle? in resource usage it says 'Two GPIO for D+/D- (Series 22ohm resistors are better)' do I have to connect these in between the usbafemale and gpio pins?

That helps. Not 100% required to work, but is good to put in

>  also on this other persons pico https://github.com/No0ne/ps2x2pico they use bi-directional level shifter

Definitely not needed.

**Reply by AbhishekAKA47** [reply]
I did the part of connecting the pico to a usbasocket as the image. I thought I got how to upload the code but looks like I just got the device_info.ino part . so i downloaded the pico-setup-windows-x64-standalone and followed a tutorial but they all end at dragging and dropping a blink.uf2 file and don't tell anything more than that. how do I upload the code to the pico using only arduino ide ? and which code do i even upload? because I don't know that as well

**Reply by AbhishekAKA47** [reply]
i have all the files installed but dont know how to open pico pio usb in this . pico - visual studio code is open everything is configured and built but dont know how to move the pico pio usb files into this
